solidity-plugin/src/main/java/org/sonarsource/solidity/SoliditySensor.java

```diff
@@ -71,14 +86,31 @@ public void analyzeFiles(SensorContext context, List<InputFile> files) {
       lastAnalyzedFile = file.toString();
       if (inSonarQube(context)) {
         System.out.println("Analyzing: " + lastAnalyzedFile);
+        getSyntaxHighlighting(context, file).save();
+      }
+    }
+  }
 
-        saveHighlighting(context, file);
+  public NewHighlighting getSyntaxHighlighting(SensorContext context, InputFile inputFile) {
+    this.highlighting = context.newHighlighting().onFile(inputFile);
+    try {
+      SolidityParser parser = Utils.returnParserUnitFromParsedFile(inputFile.contents());
+      for (Token t : parser.comments) {
+        if (t.getType() == 118) {
+          highlightComment(t);
+        } else { // TODO Structured Comments
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=14&issues=AWQCVB8xuoX1TGVkLMgu&open=AWQCVB8xuoX1TGVkLMgu)

[ISSUE_KEY]: AWQCVB8xuoX1TGVkLMgu